jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-elytron.sh

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/env bash
+
+source "${JBOSS_HOME}/bin/launch/logging.sh"
+
+function unset_kie_security_auth_env() {
+    # please keep these in alphabetical order
+    unset AUTH_LDAP_ALLOW_EMPTY_PASSWORDS
+    unset AUTH_LDAP_BASE_CTX_DN
+    unset AUTH_LDAP_BASE_FILTER
+    unset AUTH_LDAP_BIND_CREDENTIAL
+    unset AUTH_LDAP_BIND_DN
+    unset AUTH_LDAP_DEFAULT_ROLE
+    unset AUTH_LDAP_MAPPER_KEEP_MAPPED
+    unset AUTH_LDAP_MAPPER_KEEP_NON_MAPPED
+    unset AUTH_LDAP_NEW_IDENTITY_ATTRIBUTES
+    unset AUTH_LDAP_LOGIN_FAILOVER
+    unset AUTH_LDAP_LOGIN_MODULE
+    unset AUTH_LDAP_RECURSIVE_SEARCH
+    unset AUTH_LDAP_REFERRAL_MODE
+    unset AUTH_LDAP_ROLE_ATTRIBUTE_ID
+    unset AUTH_LDAP_ROLE_FILTER
+    unset AUTH_LDAP_ROLE_RECURSION
+    unset AUTH_LDAP_ROLES_CTX_DN
+    unset AUTH_LDAP_SEARCH_TIME_LIMIT
+    unset AUTH_LDAP_URL
+    unset AUTH_ROLE_MAPPER_ROLES_PROPERTIES
+}
+
+function prepareEnv() {
+    unset KIE_ELYTRON_FS_PATH
+    unset KIE_GIT_CONFIG_PATH
+    unset_kie_security_auth_env
+}
+
+function configureEnv() {
+    configure
+}
+
+function configure() {
+    configure_kie_fs_realm
+    configure_business_central_kie_git_config
+    configure_elytron_ldap_auth
+    configure_elytron_http_auth_factory
+    configure_elytron_role_mapping
+    configure_ldap_sec_domain
+    configure_new_identity_attributes
+    configure_role_decoder
+    update_activemq_domain
+    update_jboss_web_xml
+    update_security_domain
+}
+
+function configure_kie_fs_realm() {
+    local path=$(get_kie_fs_path)
+    local fs_realm="<filesystem-realm name=\"KieFsRealm\">\n\
+                    <file path=\"${path}\"/>\
+                </filesystem-realm>"
+
+    sed -i "s|<!-- ##KIE_FS_REALM## -->|${fs_realm}|" $CONFIG_FILE
+    if [ "${JBOSS_PRODUCT}" = "rhpam-kieserver" ]; then
+        JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.kie.server.services.jbpm.security.filesystemrealm.folder-path=${path}"
+    else
+        JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.uberfire.ext.security.management.wildfly.filesystem.folder-path=${path}"
+        JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.uberfire.ext.security.management.wildfly.cli.folderPath=${path}"
```

Comment:
These properties are not needed for Kie Server, but only for all other images (including dm kie server)?
If there properties are needed only for some images, it would be better to use else if or switch.

Reply:
the first system property will be added only on kieserver images, pam or dm.
the other one for all JBoss EAP based images that's rely on this configuration script.

Reply:
I think the second two are needed only for business and decision central, @Rikkola can you please confirm?

Reply:
> the first system property will be added only on kieserver images, pam or dm. the other one for all JBoss EAP based images that's rely on this configuration script.

on line 60 there is check only for pam, so there should be also add dm kie server


Yes, probably second one is only needed for images with business central as properties are for uberfire. In that case it would make sense to use else if to not have them on other images, like smartrouter.

Reply:
oh I see, didn't notice that, thanks man!, will update now.